### ISSUE TITLE
face_losses.py

### ISSUE THREAD

**Issue by MingzhenShao** [auroua/InsightFace_TF#5]
Maybe there are something mistaken in cosineface_losses()?
`output = tf.add(s * tf.multiply(cos_t, inv_mask), s * tf.multiply(cos_t_m, mask), name='cosineface_loss_output')`
But in AMSoftmax, `if(top_data[i * dim + gt] > -bias_) top_data[i * dim + gt] += bias_;` Not all the value subtract the margin.

**Comment by auroua** [comment]
I didn't reference the code of this paper. I write the method according to the paper, so there may be some mistakes. I haven't trained the model using this loss function. If you get good results in the modified form, please tell me and I will update the loss function. Thanks!
